tests/phpunit/tests/query.php

```diff
@@ -897,4 +897,71 @@ public function test_query_tag_404_does_not_throw_warning() {
 		$this->assertFalse( $q->is_tax() );
 		$this->assertFalse( $q->is_tag( 'non-existent-tag' ) );
 	}
+
+	/**
+	 * Test if $before_loop is true before loop.
+	 *
+	 * @ticket 58211
+	 */
```

Comment:
Hmm, not sure if this one technically covers `WP_Query::get_posts()` (although it would need to not use the helper method and instead make the call itself), or if it should have `@coversNothing`.

In this instance, I usually ask myself WWJD? So: @jrfnl, if you have time, what would you advise here? 🙂

Reply:
@costdev these( this and next 2) tests only checks for state of newly introduced class attribute `$before_loop` under different conditions. Logic has nothing to do specifically with `WP_Query::get_posts()`, so we can't have it there. And, as `@covers` is used only for method, we had to keep it blank. I feel we should use `@coversNothing`, but I am not 100% sure. cc: @felixarntz

Reply:
@costdev Really all that this test covers is that the initial value (i.e. the default value) of `WP_Query::$before_loop` is `true`. So no specific method is being covered here.